luthier/src/test/groovy/com/yahoo/bard/webservice/config/luthier/ResourceNodeSupplierTest.groovy

```diff
@@ -0,0 +1,52 @@
+package com.yahoo.bard.webservice.config.luthier
+
+import com.fasterxml.jackson.databind.node.ArrayNode
+import com.fasterxml.jackson.databind.node.ObjectNode
+import spock.lang.Specification
+
+class ResourceNodeSupplierTest extends Specification {
+    ResourceNodeSupplier defaultResourceNodeSupplier
+    ResourceNodeSupplier testResourceNodeSupplier
+    void setup() {
+        defaultResourceNodeSupplier = new ResourceNodeSupplier(null)
+        testResourceNodeSupplier = new ResourceNodeSupplier("DimensionConfig.json")
+    }
+
+    void cleanup() {
+
+    }
+
+    def "Node Existence"() {
```

Comment:
This can be named better. Something like:
```groovy
def "ResourceNodeSupplier returns a node if exists, and null if it doesn't.
```
Though that does make me wonder if maybe we should have the `ResourceNodeSupplier::get`
return an `Optional<ObjectNode>` if we expect `null` to be a well-defined return.

Reply:
Changed to a more current version and added another test to address the current behavior. 

Right now it throws an error in the case that { the file does not exist, the file is malformed}, instead of returning a null. 

I agree with the optional wrapper, but I am not sure if `null` will be actually meaningful in practice. (The only reasonable case to return a `null` is when `ResourceNodeSupplier( null )` is used to initialize the resourceNodeSupplier, thus the getter function see that the `resourceName == null`.) 

I suggest we should consider the optional if we plan to make a setter available for `resourceName`, which might be exposed to `null` value.